### PR TITLE
#154953736 Feature filter Ingredients and Exercise by language

### PR DESCRIPTION
**Pull request by sir3n-sn**
#### What does this PR do?
  - Add Language filter toggle to filter the enormous 7000+ Ingredients and also Exercises 
#### Description of Task to be completed?
  - When a user is logged in his current language is used to filter for the Ingredients and Exercises and 
     which is then presented to him. The only way to change this is by changing the entire webapp 
     language so as to filter. This is even more daunting when you realize that when a language does not 
     have any ingredients , over 7000+ Ingredients from the US department of  agriculture are rendered.
     Therefore this PR seeks to make this easier by presenting a simple search toggle to only filter out 
      Ingredients one needs based on its languages.
#### Any background context you want to provide?
  - To render this for the exercises the cache previous exercises functionality together with its tests as 
     this returns a cached copy of the previous ingredients rather than filtering. 
#### What are the relevant pivotal tracker stories?
   - 154953736

### CODE REVIEW

File: wger/exercises/tests/test_categories.py [hunk 1]
```diff
@@ -81,49 +81,49 @@ class AddExerciseCategoryTestCase(WorkoutManagerAddTestCase):
...
-            get_template_cache_name('exercise-overview', 2))
-        new_exercise_overview_mobile = cache.get(
-            get_template_cache_name('exercise-overview-mobile', 2))
-
-        if not self.is_mobile:
-            self.assertNotEqual(old_exercise_overview, new_exercise_overview)
-        else:
-            self.assertNotEqual(old_exercise_overview_mobile,
-                                new_exercise_overview_mobile)
+# Disabled cache in exercise view for filtering purposes so all cache tests are disabled also
```
Comment (collin5) [inline wger/exercises/tests/test_categories.py]: I don't think to comment out failing test cases is the right thing to do.

File: wger/exercises/tests/test_exercise.py [hunk 2]
```diff
@@ -422,94 +422,95 @@ class DeleteExercisesTestCase(WorkoutManagerDeleteTestCase):
...
-            get_template_cache_name('exercise-overview-mobile', 2))
-
-        if not self.is_mobile:
-            self.assertNotEqual(old_exercise_bg, new_exercise_bg)
-            self.assertNotEqual(old_exercise_overview, new_exercise_overview)
-            self.assertNotEqual(old_muscle_overview, new_muscle_overview)
-        else:
-            self.assertNotEqual(old_exercise_overview_mobile,
-                                new_exercise_overview_mobile)
+# These tests are now disabled as the caching feature is now disabled.
```
Comment (collin5) [inline wger/exercises/tests/test_exercise.py]: Same here

File: wger/exercises/views/exercises.py [hunk 3]
```diff
@@ -54,13 +54,22 @@ class ExerciseListView(ListView):
...
+        return super().dispatch(*args, **kwargs)
+
     def get_queryset(self):
         """
         Filter to only active exercises in the configured languages
         """
-        languages = load_item_languages(LanguageConfig.SHOW_ITEM_EXERCISES)
-        return Exercise.objects.accepted() \
-            .filter(language__in=languages) \
+        # TODO WRITE LOGIC FOR EXERCISES
```
Comment (collin5) [inline wger/exercises/views/exercises.py]: Could you please explain what's the reason behind this todo comment, `"WRITE LOGIC FOR EXERCISES"` seems too abstract.

File: wger/exercises/tests/test_categories.py (hunk 1 above)
Comment (sir3n-sn) [reply wger/exercises/tests/test_categories.py]: I dropped the caching functionality from the exercises as is when you filter the caching feature returns a cached copy of the previous exercises. So the test are not functional @collin5

File: wger/exercises/tests/test_exercise.py (hunk 2 above)
Comment (sir3n-sn) [reply wger/exercises/tests/test_exercise.py]: Same here @collin5

File: wger/exercises/views/exercises.py (hunk 3 above)
Comment (sir3n-sn) [reply wger/exercises/views/exercises.py]: Sorry I already did that 
Must have forgotten to remove the comment
Fixin it Asap

File: wger/exercises/tests/test_categories.py (hunk 1 above)
Comment (collin5) [reply wger/exercises/tests/test_categories.py]: @sir3n-sn Could you maybe just modify the caching logic to un-cache the page once the language is different from the previous. Coz we can't just get rid of the caching functionality.

Comment (sir3n-sn) [reply wger/exercises/tests/test_categories.py]: On it

**Pull request merged**